lib/install.js

```diff
@@ -153,6 +156,9 @@ function install(opts, cb) {
  * @returns {undefined}
  */
 function reloadProxy(opts, cb) {
+    // TODO: support hot reloading for require(esm)
+    // the challenge here is that node doesn't provide a mechanism to clear
+    // the import cache, see https://github.com/nodejs/node/issues/49442
```

Comment:
It never will AFAICT. The way forward here is supposed to be loader hooks I think. That will not be easy with the current approach I think, so might not be as much of a "TODO" as a "deprecate this feature as a part of enroute and move this concern higher in the stack".